### DISCUSSION TITLE
IJobManager.GetJob(identifier) returns a JobInfo object even when a job is not registered

### DISCUSSION THREAD

**Discussion by vppetrov** [shinyorg/shiny#866] [Q&A] [answered]
On iOS, using DryIoc (not static classes), when calling IJobManager.GetJob(identifier) the method always returns a non-null value, even though the job is not registered (happens with any random identifier). Calling IJobManager.GetJobs() returns correctly an enumeration which does not contain the job as it is not registered.

Is this behavior by design? What would be the correct way to check if a job is registered?

**Comment by aritchie** [comment, marked as answer]
All enumerations will return empty as opposed to null.  This is a standard workflow for all of .NET. 

Please ensure you review the source code in Shiny before asking tons of questions.  These forums aren't meant as general technical support.

**Reply by vppetrov** [reply]
The question is not about the enumerations (they work correctly) but about GetJob(string identifier). It returns a non-null JobInfo object, even though the job is not registered.

PS: Is there a way to get (paid) support by you? I would like to support your work.

**Reply by aritchie** [reply]
If GetJob is returning a value, the job is registered.  You can see the checks here: https://github.com/shinyorg/shiny/blob/master/src/Shiny.Jobs/AbstractJobManager.cs#L89

You can become a github sponsor and I do accept more questions on here before pushing back.  At certain levels, you get access to SponsorConnect which has a discord channel.  You can find more info about it here: https://github.com/shinyorg/shiny#readme

If you require more real-time/bugfix support, I do offer consulting which is charge on a time and materials basis.  That's usually for enterprise type setups though.
